lib/language_pack/ruby.rb

```diff
@@ -29,7 +29,7 @@ def self.use?
   end
 
   def self.bundler
-    @bundler ||= LanguagePack::Helpers::BundlerWrapper.new.install
+    @@bundler ||= LanguagePack::Helpers::BundlerWrapper.new.install
```

Comment:
Think i figured out what's going on. You're right. We detect which pack to use

https://github.com/heroku/heroku-buildpack-ruby/blob/e99ccea2cf3343f4d8ef7ffc813f754c3d5c090d/bin/compile#L12-L16

To do that we loop through all the packs and call detect.

https://github.com/heroku/heroku-buildpack-ruby/blob/e99ccea2cf3343f4d8ef7ffc813f754c3d5c090d/lib/language_pack.rb#L16-L20

To do that for Rails we have to use the bundler lockfile parser from bundler https://github.com/heroku/heroku-buildpack-ruby/blob/e99ccea2cf3343f4d8ef7ffc813f754c3d5c090d/lib/language_pack/rails4.rb#L12-L16

Since the `self.class` is different for each subclass instance it would instantiate and create a new bundler wrapper instance for each detect. Good catch, let's leave that in there. Thanks!